### PR TITLE
[AIRFLOW-500] For github enterprise auth use unique id for teams instead of slug

### PR DESCRIPTION
**Pull request by mylons**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- _[AIRFLOW-500](https://issues.apache.org/jira/browse/AIRFLOW-500)_

Testing Done:
- ran unit tests in master vs my branch. no new errors
- tested login on our internal GHE deployment

### CODE REVIEW

**Comment by bolkedebruin** [comment]
@mylons Thanks! Does this fix the security issue we have outstanding with the GH integration? **Update: never mind, it does that**. Would you mind updating the commit message/title per guidelines? Especially with security issues it is nice to have a good description.

 cc @mistercrunch

**Comment by mylons** [comment]
@bolkedebruin I tried to adhere to the guidelines, what did I miss? Does the title of the PR have to match the Jira issue title? Same with commit?

File: docs/security.rst [hunk 1]
```diff
@@ -240,7 +240,7 @@ your GHE installation will be able to login to Airflow.
     client_id = oauth_key_from_github_enterprise
     client_secret = oauth_secret_from_github_enterprise
     oauth_callback_route = /example/ghe_oauth/callback
-    allowed_teams = example_team_1, example_team_2
+    allowed_teams = 1,345,23
```
Comment (ldct) [inline docs/security.rst]: put spaces after commas

File: airflow/contrib/auth/backends/github_enterprise_auth.py [hunk 2]
```diff
@@ -154,9 +154,10 @@ def ghe_team_check(self, username, ghe_token):
                     resp.status if resp else 'None'))
 
         for team in resp.data:
-            # team json object has a slug cased team name field aptly named
-            # 'slug'
-            if team['slug'] in teams:
+            # mylons: teams are part of organizations. organizations are unique, but teams are not
+            # therefore 'slug' for a team is not necessarily unique.
+            # use id instead
```
Comment (ldct) [inline airflow/contrib/auth/backends/github_enterprise_auth.py]: this comment doesn't really make sense without the context that the following line used to be `team['slug']`

**Review by ldct — CHANGES_REQUESTED** [review]

**Comment by ldct** [comment]
this will break existing installations which place slugs into `allowed_teams`... should be introduce `allowed_team_ids` and deprecate `allowed_teams` instead?

**Comment by mylons** [comment]
@zodiac good point, but does it make sense to allow the security issue to persist?

**Comment by ldct** [comment]
@mylons ah, I missed that this affects security. You're right, we shouldn't allow slugs at all then. Is it possible to throw some friendly error message for existing users?

**Comment by mylons** [comment]
@zodiac that seems reasonable. will add that and address the other issues in the upcoming commit

**Comment by bolkedebruin** [comment]
@mylons any update? Would be nice to have it in as we can update our forthcoming apache-incubating release with it (1.7.1.3+security+compliance)

**Comment by mylons** [comment]
@bolkedebruin should i rebase again or ?

**Comment by ldct** [comment]
👍

**Comment by mylons** [comment]
this is good now -- the id being an integer is the _first_ thing I thought of when I woke up this morning, and really hoping this PR wasn't merged, haha.

File: airflow/contrib/auth/backends/github_enterprise_auth.py [hunk 3]
```diff
@@ -138,13 +138,21 @@ def get_ghe_user_profile_info(self, ghe_token):
 
     def ghe_team_check(self, username, ghe_token):
         try:
-            teams = [team.strip()
-                     for team in
-                     get_config_param('allowed_teams').split(',')]
+            # the response from ghe returns the id of the team as an integer
+            try:
+                teams = [int(team.strip())
```
Comment (ldct) [inline airflow/contrib/auth/backends/github_enterprise_auth.py]: nit: would be clearer to name this `allowed_teams`

**Review by ldct — CHANGES_REQUESTED** [review]

**Comment by ldct** [comment]
👍  again

File: airflow/contrib/auth/backends/github_enterprise_auth.py (hunk 3 above)
Comment (mylons) [reply airflow/contrib/auth/backends/github_enterprise_auth.py]: true.

**Comment by ldct** [comment]
1 more nit: capitalize first character of commit message and PR

**Comment by bolkedebruin** [comment]
LGTM

**Pull request closed without merging**